diff --git a/build/pkgs/singular/spkg-configure.m4 b/build/pkgs/singular/spkg-configure.m4
index 7cf2672836..af2eb85414 100644
--- a/build/pkgs/singular/spkg-configure.m4
+++ b/build/pkgs/singular/spkg-configure.m4
@@ -48,8 +48,8 @@ SAGE_SPKG_CONFIGURE([singular], [
                            ], [
                             AC_MSG_RESULT(no)
                             sage_spkg_install_singular=yes
-                          ])
-                        ])
+                          ], [AC_MSG_RESULT(yes)])
+                        ], [AC_MSG_RESULT(yes)])
 
                       AC_LANG_POP()
                       LIBS="${ORIG_LIBS}"
