From f87cf6c523f38b4b4d1d4a6c467a67db37a2b75f Mon Sep 17 00:00:00 2001
From: Ralf Stephan <ralf@ark.in-berlin.de>
Date: Mon, 15 Jan 2018 09:00:51 +0100
Subject: [PATCH] 24411: adapt to pynac-0.7.14

---
 src/sage/functions/gamma.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sage/functions/gamma.py b/src/sage/functions/gamma.py
index fe25c31d2a..635cc77d6e 100644
--- a/src/sage/functions/gamma.py
+++ b/src/sage/functions/gamma.py
@@ -163,7 +163,7 @@ class Function_gamma(GinacFunction):
             :meth:`gamma`
         """
         GinacFunction.__init__(self, 'gamma', latex_name=r"\Gamma",
-                               ginac_name='tgamma',
+                               ginac_name='gamma',
                                conversions={'mathematica':'Gamma',
                                             'maple':'GAMMA',
                                             'sympy':'gamma',
-- 
2.16.2

