diff --git a/src/doc/en/introspect/__init__.py b/src/doc/en/introspect/__init__.py
deleted file mode 100644
index 8b13789..00000000
--- a/src/doc/en/introspect/__init__.py
+++ /dev/null
@@ -1 +0,0 @@
-
diff --git a/src/doc/en/introspect/conf.py b/src/doc/en/introspect/conf.py
deleted file mode 100644
index 4c63267..00000000
--- a/src/doc/en/introspect/conf.py
+++ /dev/null
@@ -1,21 +0,0 @@
-# -*- coding: utf-8 -*-
-# Sage introspection build configuration file.
-# See sagenb.notebook.cell.Cell.set_introspect_html() for details.
-
-import sys, os
-from sage.env import SAGE_DOC_SRC
-sys.path.append(SAGE_DOC_SRC)
-from common.conf import *
-
-extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo',
-              'sphinx.ext.extlinks']
-
-templates_path = ['templates']
-html_static_path = ['static']
-
-html_use_modindex = False
-html_use_index = False
-html_split_index = False
-html_copy_source = False
-
-todo_include_todos = True
diff --git a/src/doc/en/introspect/static/empty b/src/doc/en/introspect/static/empty
deleted file mode 100644
index 8b13789..00000000
--- a/src/doc/en/introspect/static/empty
+++ /dev/null
@@ -1 +0,0 @@
-
diff --git a/src/doc/en/introspect/templates/layout.html b/src/doc/en/introspect/templates/layout.html
deleted file mode 100644
index 2e17d5b..00000000
--- a/src/doc/en/introspect/templates/layout.html
+++ /dev/null
@@ -1,3 +0,0 @@
-<div class="docstring">
-    {% block body %} {% endblock %}
-</div>
diff --git a/src/sage/misc/sphinxify.py b/src/sage/misc/sphinxify.py
index 4270294..fa62413 100644
--- a/src/sage/misc/sphinxify.py
+++ b/src/sage/misc/sphinxify.py
@@ -26,7 +26,6 @@ import os
 import re
 import shutil
 from tempfile import mkdtemp
-from sage.env import SAGE_DOC_SRC
 from sphinx.application import Sphinx
 
 
@@ -82,12 +81,35 @@ def sphinxify(docstring, format='html'):
     with open(rst_name, 'w') as filed:
         filed.write(docstring)
 
-    confdir = os.path.join(SAGE_DOC_SRC, 'en', 'introspect')
-
-    open(os.path.join(srcdir, 'docutils.conf'), 'w').write(r"""
+    confdir = os.path.join(srcdir, 'en' , 'introspect')
+    os.makedirs(confdir)
+    with open(os.path.join(confdir, 'conf.py'), 'w') as filed:
+        filed.write(r"""
+extensions = ['sphinx.ext.autodoc', 'sphinx.ext.mathjax', 'sphinx.ext.todo', 'sphinx.ext.extlinks']
+
+templates_path = ['templates']
+html_static_path = ['static']
+
+html_use_modindex = False
+html_use_index = False
+html_split_index = False
+html_copy_source = False
+
+todo_include_todos = True""")
+    templatesdir = os.path.join(confdir, 'templates')
+    os.makedirs(templatesdir)
+    with open(os.path.join(templatesdir, 'layout.html'), 'w') as filed:
+        filed.write(r"""<div class="docstring">
+    {% block body %} {% endblock %}
+</div>""")
+    staticdir = os.path.join(confdir, 'static')
+    os.makedirs(staticdir)
+    with open(os.path.join(staticdir, 'empty'), 'w') as filed: pass
+
+    with open(os.path.join(srcdir, 'docutils.conf'), 'w') as filed:
+        filed.write(r"""
 [parsers]
-smart_quotes = no
-""")
+smart_quotes = no""")
     doctreedir = os.path.join(srcdir, 'doctrees')
     confoverrides = {'html_context': {}, 'master_doc': 'docstring'}
 
