From 8e6b2d3219b19201bb8c02ff91e1367100462593 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Fri, 3 Feb 2017 16:07:08 +0530
Subject: [PATCH] Fix cython bug

---
 src/setup.py | 1 +
 1 file changed, 1 insertion(+)

--- a/src/setup.py
+++ b/src/setup.py
@@ -367,6 +367,7 @@
             compiler_directives=self.cython_directives,
             compile_time_env=self.compile_time_env,
             create_extension=self.create_extension,
+            include_path=sys.path,
             # Debugging
             gdb_debug=self.debug,
             output_dir=os.path.join(self.build_lib, "sage"),
