--- a/src/sage/misc/lazy_import_cache.py
+++ b/src/sage/misc/lazy_import_cache.py
@@ -5,6 +5,7 @@
 """
 
 import os
+import hashlib
 
 from ..env import SAGE_SRC, DOT_SAGE
 
@@ -31,5 +32,7 @@
         sage: sage.misc.lazy_import_cache.DOT_SAGE = OLD
     """
     mangled = os.path.realpath(SAGE_SRC).replace(os.sep, '_')
+    if len(mangled) > 255 - 25:
+        mangled = hashlib.sha256(mangled.encode('utf-8')).hexdigest()
     return os.path.join(DOT_SAGE, 'cache',
                         "%s-lazy_import_cache.pickle" % mangled)
